libraries/freertos_plus/standard/freertos_plus_tcp/source/portable/NetworkInterface/esp32/NetworkInterface.c

```diff
@@ -44,8 +44,13 @@ enum if_state_t {
 static const char *TAG = "NetInterface";
 volatile static uint32_t xInterfaceState = INTERFACE_DOWN;
 
+/* protect the function declaration itself instead of using
+   #if everywhere.                                        */
 #if ( ipconfigHAS_PRINTF != 0 )
     static void prvMonitorResources();
+    #define PRINT_RESOURCE_LOGS prvMonitorResources();
+#else
+    #define PRINT_RESOURCE_LOGS
 #endif
```

Comment:
How about instead of this you leave the code as just `prvMonitorResources(); ` where it is being called, and here you do something like this.

```
#if ( ipconfigHAS_PRINTF != 0 )
    static void prvMonitorResources();	    static void prvMonitorResources();
#else
    #define prvMonitorResources()
#endif
```	

That way you can also remove the implementation of the function itself with the same guard on ipconfigHAS_PRINTF

Reply:
Done